### ISSUE TITLE
any matcher question

### ISSUE THREAD

**Issue by JRealFu** [ulfurinn/wongi-engine#66]
Hi, 

I'm trying to use the any matcher to  include "or" conditions, but the engine only matches if the the first option rule is verified.

- Case 1

```
engine = Wongi::Engine.create
ruleset "my Rules" do
    rule "error en rojo" do
        forall {
            any {
                option {
                    has :A, :scenario, "OT en rojo"
                }    
                option {
                    has :A, :msisdn, "12345678"
                }    
            }
            has :A, :cliente, "manolito" 
        }
        make {
            action { 
              puts "Es un error en rojo"
            }
        }

    end    
end    
engine << Wongi::Engine::Ruleset["my Rules"]
engine << ["123456", :msisdn, "12345678"]
engine << ["123456", :cliente, "manolito"]
engine << ["123456", :scenario, "OT en rojo"]
```

**Output:  Es un error en rojo**

 

- Case 2

```
engine = Wongi::Engine.create
ruleset "my Rules" do
    rule "error en rojo" do
        forall {
            any {
                option {
                    has :A, :scenario, "OT en rojo"
                }    
                option {
                    has :A, :msisdn, "12345678"
                }    
            }
            has :A, :cliente, "manolito" 
        }
        make {
            action { 
              puts "Es un error en rojo"
            }
        }

    end    
end    
engine << Wongi::Engine::Ruleset["my Rules"]
engine << ["123456", :msisdn, "12345678"]
engine << ["123456", :cliente, "manolito"]
engine << ["123456", :scenario, "OT en rojos"]
```

**Output:**

**Comment by ulfurinn** [comment]
Hm, yes. Until I fix this, the workaround is to put the `has` clause before the `any` clause.

**Comment by ulfurinn** [comment]
Released v0.3.1.

This was in a surprisingly fundamental part to have gone undetected by anyone for so long.

**Comment by JRealFu** [comment]
Thank for your reply.

My WA was create a custom matcher whihc uses regexp, so I can validate several values

        forall {
            has :A, :msisdn, :B
            assert { |token|

                /hola|adios/.match(token[:B])
            }
        }

**Comment by Ceddy** [comment]
I actually have a question about this particular use case, which is not addressed by the latest fix (which might be because I'm using this incorrectly).

So I have 2 options but I want both to bind to the same name (this might be incorrect?)
```
    let :production do
      engine << rule do
        forall {
          any {
            option {
              has :A, :path1, :PathVar
            }
            option {
              has :A, :path2, :PathVar
            }
          }
        }
        make {
          collect :PathVar, :paths
        }
      end
    end
```

So my tests look like this:
```
    it 'should fire on the first path' do
      engine << [:x, :path1, true]
      expect(production.tokens).to have(1).item
      expect(engine.collection(:paths)).to include(true)
    end

    it 'should fire on the second path' do
      engine << [:x, :path2, true]
      expect(production.tokens).to have(1).item
      expect(engine.collection(:paths)).to include(true)
    end
```

but basically whichever option path is included first passes, and the second fails. Is this expected? What I want is to be able to optionally match on 2 different things but name them the same, so I can use the variable in a different assign block below.

Thanks

**Comment by ulfurinn** [comment]
Released 0.3.2.
